pkg/fetch/metrics_fetcher.go

```diff
@@ -57,36 +57,62 @@ func NewMetricsFetcher(fetcher Fetcher, clock clock.Clock, name string) Fetcher
 		fetcher: fetcher,
 		clock:   clock,
 
-		fetchBlobBlobSizeBytes:        httpFetcherOperationsBlobSizeBytes.WithLabelValues(name, "Fetch Blob"),
+		fetchBlobBlobSizeBytes:        httpFetcherOperationsBlobSizeBytes.MustCurryWith(map[string]string{"name": name, "operation": "Fetch Blob"}),
 		fetchBlobDurationSeconds:      blobAccessOperationsDurationSeconds.MustCurryWith(map[string]string{"name": name, "operation": "FetchBlob"}),
 		fetchDirectoryDurationSeconds: blobAccessOperationsDurationSeconds.MustCurryWith(map[string]string{"name": name, "operation": "FetchDirectory"}),
 	}
 }
 
-func (mf *metricsFetcher) updateDurationSeconds(vec prometheus.ObserverVec, code codes.Code, timeStart time.Time) {
-	vec.WithLabelValues(code.String()).Observe(mf.clock.Now().Sub(timeStart).Seconds())
+func (mf *metricsFetcher) updateDurationSeconds(vec prometheus.ObserverVec, code codes.Code, timeStart time.Time, qualifiers []*remoteasset.Qualifier) {
+	if len(qualifiers) == 0 {
+		vec.WithLabelValues(code.String(), "N/A").Observe(mf.clock.Now().Sub(timeStart).Seconds())
+	} else {
+		resourceType := "N/A"
+		for _, qualifier := range qualifiers {
+			if qualifier.Name == "resource_type" {
+				resourceType = qualifier.Value
+				break
+			}
+		}
+		vec.WithLabelValues(code.String(), resourceType).Observe(mf.clock.Now().Sub(timeStart).Seconds())
```

Comment:
Can this be simplified to avoid the if statement all together? Since we're using `resourceType := "N/A"` at the start anyway we should be getting the same result if qualifiers is empty.